src/pages/admin/financials/edit.tsx

```diff
@@ -84,87 +89,59 @@ interface AdminEditFinancialsProps extends AppProps {
   const [withdrawlLoading, setWithdrawlLoading] = useState(false)
 
-  const payoutFunds = async () => {
+  const claimFunds = async (ah: AuctionHouse) => {
     if (!publicKey || !signTransaction || !wallet) {
       toast.error('Wallet not connected')
-
       login()
-
       return
     }
 
-    const auctionHouse = new PublicKey(marketplace.auctionHouse.address)
-    const authority = new PublicKey(marketplace.auctionHouse.authority)
-    const treasuryMint = new PublicKey(marketplace.auctionHouse.treasuryMint)
-    const auctionHouseTreasury = new PublicKey(
-      marketplace.auctionHouse.auctionHouseTreasury
-    )
-
-    const treasuryWithdrawalDestination = new PublicKey(
-      marketplace.auctionHouse.treasuryWithdrawalDestination
-    )
-
-    const auctionHouseTreasuryBalance = await connection.getBalance(
-      auctionHouseTreasury
-    )
-
-    const withdrawFromTreasuryInstructionAccounts = {
-      treasuryMint,
-      authority,
-      treasuryWithdrawalDestination,
-      auctionHouseTreasury,
-      auctionHouse,
-    }
-    const withdrawFromTreasuryInstructionArgs = {
-      amount: auctionHouseTreasuryBalance,
-    }
-
-    const withdrawFromTreasuryInstruction =
-      createWithdrawFromTreasuryInstruction(
-        withdrawFromTreasuryInstructionAccounts,
-        withdrawFromTreasuryInstructionArgs
-      )
-
-    const txt = new Transaction()
-
-    txt.add(withdrawFromTreasuryInstruction)
-
-    txt.recentBlockhash = (await connection.getRecentBlockhash()).blockhash
-    txt.feePayer = publicKey
-
-    let signed: Transaction | undefined = undefined
-
-    try {
-      signed = await signTransaction(txt)
-    } catch (e: any) {
-      toast.error(e.message)
-      return
-    }
+    toast('Sending the transaction to Solana.')
+    setWithdrawlLoading(true)
+    await sdk.claimFunds(ah)
+    toast.success('The transaction was confirmed.')
+    setWithdrawlLoading(false)
+  }
 
-    let signature: string | undefined = undefined
+  const tokens = marketplace.auctionHouses?.map(({ treasuryMint }) =>
+    tokenMap.get(treasuryMint)
+  )
 
-    try {
-      toast('Sending the transaction to Solana.')
-      setWithdrawlLoading(true)
-      signature = await connection.sendRawTransaction(signed.serialize())
+  // DUMMY TOKENS FOR TESTING
+  // const tokens = [
+  //   tokenMap.get('So11111111111111111111111111111111111111112'),
+  //   tokenMap.get('EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'),
+  // ]
 
-      await connection.confirmTransaction(signature, 'confirmed')
+  const treasuryBalanceMap = useMemo(() => {
+    const map = new Map()
 
-      toast.success('The transaction was confirmed.')
-    } catch (e) {
-      toast.error(e.message)
-    }
-    setWithdrawlLoading(false)
-  }
+    marketplace.auctionHouses?.forEach(async (ah) => {
+      const balance = await getTreasuryBalance(ah, connection)
+      map.set(ah.treasuryMint, balance)
+    })
+    return map
+  }, [connection, marketplace.auctionHouses])
```

Comment:
I believe react likes it best when async results are handled with state and useEfffect. I recommend having `const [treasuryBalances, setTreasuryBalances] = useState({})`

then instead of forEach async do a map to create a list of promises. Use promise.all so the tokens are looked up at the same time. Also ensures you get them at the same time.